src/collapse/perf.rs

```diff
@@ -154,29 +188,96 @@ impl Collapse for Folder {
     }
 }
 
-impl From<Options> for Folder {
-    fn from(mut opt: Options) -> Self {
-        opt.include_pid = opt.include_pid || opt.include_tid;
-        Self {
-            cache_line: Vec::default(),
-            event_filtering: EventFilterState::None,
-            in_event: false,
-            occurrences: FnvHashMap::with_capacity_and_hasher(512, fnv::FnvBuildHasher::default()),
-            pname: String::new(),
-            skip_stack: false,
-            stack: VecDeque::default(),
-            opt,
+impl Folder {
+    fn collapse_single_threaded<R>(&mut self, mut reader: R) -> io::Result<()>
+    where
+        R: BufRead,
+    {
+        let mut line = String::new();
```

Comment:
Should this have an initial capacity like you added to the dtrace collapser? I think either neither of them should have it (my guess), or both of them should have it.

Reply:
👍 Will go with neither.